config/m4/java.m4

```diff
@@ -0,0 +1,71 @@
+#
+# Copyright (C) Mellanox Technologies Ltd. 2001-2017.  ALL RIGHTS RESERVED.
+#
+# See file LICENSE for terms.
+#
+#
+#
+# Check for Java support
+#
+#
+java_happy="no"
+AC_ARG_WITH([java],
+            [AC_HELP_STRING([--with-java],
+                            [Compile Java UCX (default is NO).]
+                           )], [with_java=yes], [with_java=no])
+
+
+AS_IF([test "x$with_java" != xno],
+      [AS_IF([test "x$with_java" == xyes],
+             [AS_IF([test -n "$JAVA_HOME"],
+                    [],
+                    [
+                     READLINK_FKNOWN=$(readlink -f xxx 1>/dev/null 2>&1 && echo yes)
```

Comment:
why not AC_CHECK_PROG?

Reply:
what is -f xxx

Reply:
Sorry I didn't reply, been on vacation.
Changing to AC_CHECK_PROG(..., readlink, ...)